docs/source/examples/pt_zne.md

```diff
@@ -0,0 +1,386 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+```{tags} cirq, zne, pt, intermediate
+```
+# Zero-Noise Extrapolation with Pauli Twirling
+
+This tutorial explores how noise tailoring can improve the effectiveness of quantum error mitigation techniques.
+Specifically, we analyze how converting coherent noise into incoherent noise through Pauli Twirling (PT) impacts the performance of [Zero-Noise Extrapolation](../guide/zne.md) (ZNE).
+
+In this tutorial, we will:
+
+1. Define and compare coherent and incoherent noise
+2. Apply Pauli Twirling to transform coherent noise into incoherent noise
+3. Compare the performance of ZNE
+    1. on its own, and
+    2. in combination with [Pauli Twirling](../guide/pt.md)
+
+By the end of the example, you will understand when and how noise tailoring can enhance ZNE.
+
+## Coherent noise vs. Incoherent noise
+
+Noise on quantum devices can broadly categorized into two types: Coherent, Incoherent. Each has different properties that can be unfavorable toward a quantum circuit in different ways.
+
+**Coherent noise** is a reversible process as long as the noisy unitary transformation is known beforehand which is not always the case. These types of noise maintain the purity of the state. But in a quantum circuit subjected to coherent noise, the errors are easily carried across the circuit. This can be discerned through the **average gate infidelity** $r(\mathcal{E})$. When coherent errors contribute to a portion of the total error-rate, the worst case infidelity can scale as $\sqrt{r(\mathcal{E})}$ {cite}`Wallman_2014` which in turn reduces the performance of a quantum device by orders of magnitude. Thus, dealing with coherent noise requires a large resource overhead to acquire inferred knowledge of the noise unitaries which can then be used to reverse the effects.
+
+```{note}
+If $\mathcal{F}$ is the fidelity defining the success of preparing an arbitrary pure state $\rho$, then
+$1-\mathcal{F}$ is the **average gate infidelity**.
+```
+
+**Incoherent noise** is a process that results in the quantum system entangling with its environment i.e. this type of noise is irreversible. The system and the environment end up in a mixed state. It scales linearly in the small error limit. 
+The noise channel can be described using Pauli operators which makes it easy to analyze and simulate. Worst case error rate is directly proportional to the **average gate infidelity**. 
+
+For example, a depolarizing noise channel is a stochastic noise channel where a noiseless process is probabilistically mixed with orthogonal errors. If $\rho$ is a single qubit state, $p$ is the probabilistic error rate and $\mathcal{E}(\rho)$ is the noise channel:
```

Comment:
This is a great example, but it needs a connection back to (in)coherent noise. Can you add something, maybe at the end?

Do you think we should have an example for coherent noise too?

Reply:
The depolarizing noise example is to show how Pauli twirling is equivalent to tailoring the noise into a depolarizing noise channel.

Reply:
I don't quite see how this relates to pauli twirling. Can you help spell it out?

From my perspective the main idea in pauli twirling is to add gates that compile to the identity, but change the impact of noise. So when you say the depolarizing channel can be written in terms of paulis, I don't see how that is pauli twirling.

Reply:
> From my perspective the main idea in pauli twirling is to add gates that compile to the identity,

Uh, no. Pauli twirling's goal is to transform some noise channel into a depolarizing noise channel. The circuits average over to a mixed noise channel which is what I was trying to show.

Reply:
> Pauli twirling's goal is to transform some noise channel into a local depolarizing noise channel.

Right, yes, good point. That's the main goal. When adding gates surrounding each CNOT, you do it in such a way so that the CNOT is still the logical operation (up to global phase). In that sense are we not adding gates that end up compiling **out** (should you compile, which, you shouldn't)?

Reply:
I am confused about the 'compile to identity' phrase. 

If we start out with a CNOT subjected to a coherent noise channel, after Pauli twirling it's a CNOT gate with incoherent noise.

Reply:
> I am confused about the 'compile to identity' phrase.

Yeah that was sloppy phrasing on my part, sorry! I think we agree that the effect of pauli twirling on an ideal CNOT is the identity map (PT(ideal CNOT) = ideal CNOT).

> Pauli twirling is equivalent to a local depoalrizing channel.

Is that true in all cases? I think Pauli twirling (in the infinite samples limit) only guarantees a Pauli channel is spit out, right?

BTW thanks for going back and forth on this a few times. It's really helpful for my understanding!

Reply:
> I think Pauli twirling (in the infinite samples limit) only guarantees a Pauli channel is spit out, right?

Depends on the noise strength, the number of Pauli twirled circuits etc. If you look at the gif example, I ended up at a finite number of pauli twirled circuits that gave us the ideal CNOT PTM. I'll edit the lines to discuss this. 

> BTW thanks for going back and forth on this a few times. It's really helpful for my understanding!

No worries!